### PR TITLE
Badge aggregation

### PR DESCRIPTION
**Pull request by carlvitzthum**
The main work of the badge aggregation is done on the Snovault side. This branch includes schema additions, types file adjustments, tests, and a couple tweaks to search.py.

### Schema changes
- `facets_aggregated_badges` mixin was created and added to experiment sets, experiments, and biosample (including all child schemas)

### Tests
- Some badges to master-inserts
- Badges in workbook-inserts (just for ES-based tests)
- test_aggregation.py and test_types_badges.py

### Types files
- Added badge embedding to biosample, experiment, experiment sets
- Added aggregated_items information to types files. This is simply a dictionary keyed by the items (any embedded object or sub-object) that you want to aggregated. For example,
```
    aggregated_items = {
        "badges": ["message", "badge.positive_badge", "badge.warning_badge", "badge.uuid"]
    }
```
This will aggregated all "badges" items found during the embedding process and will look for the corresponding list of fields on those found items. So, the following fields would be collected at any given level during embedding for the above example: `...badges.message`, `...badges.badge.positive_badge`, etc.

### Search changes
- Do not try to find the schema for aggregated_items fields (they don't exist). This was already being done for audits
- Added support for a `weight` field that can be specified for facets. The higher the weight, the lower the facet will be displayed in the list. If a facet doesn't have a specified weight (most at this point), it gets a weight of 0. Order of these facets are conserved from the ordering in the schemas (given they are all 0). Audits are given a weight of 999. Positive badges have a weight of 997 and negative badges have a weight of 998, so that they are displayed right above audits. See [experiment search](http://fourfront-hotseat.9wzadzju3p.us-east-1.elasticbeanstalk.com/search/?type=Experiment) for an example.

### CODE REVIEW

File: src/encoded/schemas/mixins.json [hunk 1]
```diff
@@ -615,6 +615,16 @@
...
         }
+    },
+    "facets_aggregated_badges" : {
+        "aggregated_items.badges.item.badge.positive_badge": {
+            "title": "Positive Badges",
+            "weight": 997
+        },
+        "aggregated_items.badges.item.badge.warning_badge": {
+            "title": "Warning Badges",
+            "weight": 998
```
Comment (Chadius) [inline src/encoded/schemas/mixins.json]: Did you want to rename "weight" to "order"? Alex's column changes also uses "order" in the same way.

File: src/encoded/search.py [hunk 2]
```diff
@@ -777,10 +776,10 @@ def initialize_facets(types, doc_types, prepared_terms, schemas):
...
     ]
     audit_facets = [
-        ('audit.ERROR.category', {'title': 'Audit category: ERROR'}),
-        ('audit.NOT_COMPLIANT.category', {'title': 'Audit category: NOT COMPLIANT'}),
-        ('audit.WARNING.category', {'title': 'Audit category: WARNING'}),
-        ('audit.INTERNAL_ACTION.category', {'title': 'Audit category: DCC ACTION'})
+        ('audit.ERROR.category', {'title': 'Audit category: ERROR', 'weight': 999}),
+        ('audit.NOT_COMPLIANT.category', {'title': 'Audit category: NOT COMPLIANT', 'weight': 999}),
+        ('audit.WARNING.category', {'title': 'Audit category: WARNING', 'weight': 999}),
+        ('audit.INTERNAL_ACTION.category', {'title': 'Audit category: DCC ACTION', 'weight': 999})
```
Comment (Chadius) [inline src/encoded/search.py]: "weight" or "order"?

File: src/encoded/tests/test_search.py [hunk 3]
```diff
@@ -138,12 +136,18 @@ def test_search_facets_and_columns_order(workbook, testapp, registry):
     schema = type_info.schema
     schema_facets = [('type', {'title': 'Data Type'})]
     schema_facets.extend(schema['facets'].items())
-    schema_columns = [(name, obj.get('title')) for name,obj in schema['columns'].items()]
+    # the following facets are added after schema facets
+    schema_facets.append(('status', {'title': 'Status'}))
+    # remove any disabled facets
+    schema_facets = [fct for fct in schema_facets if not fct[1].get('disabled', False)]
+    sort_facets = sorted(schema_facets, key=lambda fct: fct[1].get('weight', 0))
```
Comment (Chadius) [inline src/encoded/tests/test_search.py]: Similar question about using "weight" or "order" here.

**Comment by Chadius** [comment]
Looks good to me, besides renaming weight to order.

**Comment by carlvitzthum** [comment]
Renamed "weight" to "order" for facets, per feedback from @Chadius.

**Comment by Chadius** [comment]
Sweet, thanks for renaming them.

**Pull request merged**